src/main.cpp

```diff
@@ -5435,10 +5435,13 @@ int main_map(int argc, char** argv) {
     };
 
     for (int i = 0; i < thread_count; ++i) {
-        Mapper* m;
+        Mapper* m = nullptr;
         if(xindex && gcsa && lcp) {
             // We have the xg and GCSA indexes, so use them
             m = new Mapper(xindex, gcsa, lcp);
+        } else {
```

Comment:
Yeah, I'm not sure why we're creating and tearing down the mapper all the time (or why we need a separate mapper for each thread).